### DISCUSSION TITLE
Is there an alternative to using a Splash Screen? Could I just use a custom view instead?

### DISCUSSION THREAD

**Discussion by bchehraz** [ionic-team/capacitor#5300] [Q&A] [answered]
(Related to my previous question: https://github.com/ionic-team/capacitor/discussions/5298)

To solve my particular problem of being unable to dynamically change what is displayed on the Splash Screen, couldn't I just stop using the Capacitor's Splash Screen and just have a custom view that I show while the app is initializing? Is there any issue with this? In other words, is there a compelling reason to continue using Capacitor's Splash Screen, knowing its limitations?

The answer could be an easy "yes", but...

Is there anything unique that Capacitor's Splash Screen provides that I need to be aware of when implementing my own? In other words, what could I be missing out on by doing this?

Thanks!

**Comment by jcesarmobile** [comment, marked as answer]
Yes, using the plugins provided by Capacitor team is not mandatory, you can create your own.

Things to have in mind:
- on iOS the OS first shows the launch storyboard and then we show the same storyboard again. The launch storyboard doesn't allow to run any code, so users will still first see the launch storyboard and they your custom view/storyboard
- on Android first the activity background is shown by the OS and then we show a custom view/layout that looks similar. For the initial background you won't be able to add any code neither.

So for both cases you should make a launch storyboard/background that look similar to your custom splash so the change between them is not very noticeable.

Also have in mind that if you are going to make a network request in your splash to query the company name or whatever, that also takes some time and your view won't be displayed or will be displayed before getting the response back.

**Reply by bchehraz** [reply]
Thank you for the reply. So I'm trying to figure out what these storyboards are supposed to look like... After uninstalling the splash screen plugin, it seems like on Android there's just a logo in the center of the screen whereas on iOS it's plain white screen, no logo or anything. 

It might be easier if I continue to use the Capacitor splash screen and seamlessly transition to a custom view.

I'm still experimenting a bit, I'll keep this discussion open for now.

**Reply by bchehraz** [reply]
Alright after today I went with removing the Capacitor splash screen as the default storyboards seem to be minimal and easy to transition from for both Android and iOS, was able to get a pretty neat custom loader going that shows the company name from local storage. Thanks again!
